addons/website/controllers/main.py

```diff
@@ -914,6 +914,30 @@ def theme_upload_font(self, name, data):
             - attachment id
             - attachment URL
         """
+        def check_content(filename, data):
+            """ Returns True only if data matches the font extension. """
+            # Do not pollute general guess_mimetype with this.
+            ext = filename.split('.')[-1].lower()
+            if ext == 'otf':
+                return data.startswith(b'OTTO')
+            elif ext == 'woff':
+                return data.startswith(b'wOFF')
+            elif ext == 'woff2':
+                return data.startswith(b'wOF2')
+            elif ext == 'ttf':
+                # Based on https://docs.fileformat.com/font/ttf/#true-type-file-format-specifications
+                TOC_OFFSET = 12
+                TOC_ENTRY_LENGTH = 16
+                table_size = int.from_bytes(data[4:6], 'big') * TOC_ENTRY_LENGTH
+                if TOC_OFFSET + table_size > len(data):
```

Comment:
Question: where do the offset and length values (so 12 and 16) come from ? And why does it start with an offset ? :slightly_smiling_face: (I read your link a bit but I didn't see these information right away)

Reply:
That spec if strangely ordered indeed.
So, here is my explanation:
The file begins with the following:
- bytes 0~3: uint32: scaler type
- bytes 4~5: uint16: numTables => read when computing the table_size
- bytes 6~7: uint16: searchRange
- bytes 8~9: uint16: entrySelector
- bytes 10~11: uint16: rangeShift

Then comes the first entry of the table of content at byte 12.

Each entry is then made of 4 uint32 => 16 bytes. (tag, checksum, offset, length)
We only look at the tag of each entry.